dotcom-rendering/src/components/EnhancePinnedPost.importable.tsx

```diff
@@ -124,24 +146,28 @@ export const EnhancePinnedPost = () => {
 		const config = {
 			childList: true,
 			subtree: true,
-		};
+		} satisfies MutationObserverInit;
 
 		observer.observe(pinnedPost, config);
 
 		return () => observer.disconnect();
-	}, []);
+	}, [checkContentHeight, pinnedPost]);
 
 	useEffect(() => {
-		pinnedPostCheckBox?.addEventListener('change', () =>
-			handleClickTracking(renderingTarget),
-		);
+		if (!pinnedPost) return;
+		if (!pinnedPostCheckBox) return;
 
-		return () => {
-			pinnedPostCheckBox?.removeEventListener('change', () =>
-				handleClickTracking(renderingTarget),
+		const listener = () =>
```

Comment:
Extracting this to a function ensures that the [`removeEventListener` actually works](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/removeEventListener#matching_event_listeners_for_removal).